cylc/rose/platform_utils.py

```diff
@@ -56,7 +57,12 @@ def get_platform_from_task_def(flow: str, task: str) -> Dict[str, Any]:
         workflow_id,
         WorkflowFiles.FLOW_FILE_PROCESSED,
     )
-    config = WorkflowConfig(flow, flow_file, Values())
+
+    config = WorkflowConfig(
+        flow,
+        flow_file, Values(),
+        force_compat_mode=force_compat_mode(flow_file)
```

Comment:
Note, needs matching import.

```suggestion
        force_compat_mode=get_compat_mode(get_workflow_run_dir(workflow_id))
```